docker/tarfile/src.go

```diff
@@ -35,14 +38,55 @@ type layerInfo struct {
 	size int64
 }
 
-// NewSource returns a tarfile.Source for the specified path.
-func NewSource(path string) *Source {
+// NewSourceFromFile returns a tarfile.Source for the specified path
```

Comment:
Please document that compressed input is accepted.

Reply:
fixed